lib/giraph/subquery.rb

```diff
@@ -0,0 +1,72 @@
+module Giraph
+  # Defines a thin & sane API for the sub-query extraction
+  # from the context provided to resolvers via AST, graphql-ruby API
+  # and a touch of regex.
+  class Subquery
```

Comment:
This class is kind of frightening - we should try and get the `graphql-ruby` gem to expose some of this information via its public API so we aren't tied to its internals.